pkg/generate/cspc.go

```diff
@@ -181,10 +182,14 @@ func makePools(poolType string, nDevices int, bd map[string][]v1alpha1.BlockDevi
 			})
 		}
 		return &spec, nil
-	} else if poolType == string(cstorv1.PoolMirrored) {
-		if nDevices%2 != 0 {
-			return nil, fmt.Errorf("mirrored pool requires multiples of two block device")
+	case string(cstorv1.PoolMirrored), string(cstorv1.PoolRaidz), string(cstorv1.PoolRaidz2):
+		// 1. Check for devices per node count against the fixed minimum
+		min := minCount()[poolType]
+		if nDevices < min {
+			return nil, fmt.Errorf("%s pool requires a minimum of %d block device per node",
+				poolType, min)
 		}
```

Comment:
Since `nDevices` is user input instead of validating here, better to validate  at initial phase itself [here](https://github.com/openebs/openebsctl/blob/92f3001d93c34dabea0fcfbce5ed19d80dc8266e/pkg/generate/cspc.go#L46)